opentelemetry-sdk/src/resource/mod.rs

```diff
@@ -62,6 +62,24 @@ impl Default for Resource {
 }
 
 impl Resource {
+    /// Creates a Builder that allows you to configure multiple aspects of the Resource.
+    ///
+    /// Starts with a [Resource::default()].
+    pub fn builder() -> ResourceBuilder {
```

Comment:
shouldn't we remove all other pub methods on Resource, and ask users to user ResourceBuilder only?

Reply:
I had comments on our thread asking about this to confirm understanding before making the changes since it's used in quite a few places throughout the codebase

Reply:
removed the public api options for creating resources